src/test/java/io/stargate/sgv2/jsonapi/service/shredding/DocumentShredderWithExtendedTypesTest.java

```diff
@@ -332,6 +332,31 @@ private void _testShredUUIDAutoGeneration(CollectionIdType idType, int uuidVersi
     }
   }
 
+  @Nested
+  class OkCasesBinaryVector {
+    @Test
+    public void shredSimpleBinaryVector() throws Exception {
+      final String inputJson =
+          """
+                  {
+                    "age" : 39,
+                    "$vector": {"$binary": "PoAAAD6AAAA+gAAAPoAAAD6AAAA="}
```

Comment:
I get this binary by using Java to encode a float array `float[] vector = {0.25f, 0.25f, 0.25f, 0.25f, 0.25f}`. Is this what we expect from the client?

Reply:
It is, although you  will probably want to create payload `Base64Variant.encode()` and helper methods from `Float` to construct it instead of hard-coding.

But I think even more important part is that we need Integration Tests to ensure end-to-end handling -- that is, what is sent with `insertOne()` is successfully read back as same vector.